src/Content/Version.php

```diff
@@ -109,77 +152,70 @@ public function model(): ModelWithContent
 	/**
 	 * Returns the modification timestamp of a version
 	 * if it exists
-	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 */
 	public function modified(
-		string $language = 'default'
+		Language|string $language = 'default'
 	): int|null {
-		$this->ensure($language);
-		return $this->model->storage()->modified($this->id, $language);
+		if ($this->exists($language) === true) {
+			return $this->model->storage()->modified($this->id, $this->language($language));
+		}
+
+		return null;
 	}
 
 	/**
 	 * Moves the version to a new language and/or version
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function move(string $fromLanguage, VersionId $toVersionId, string $toLanguage): void
-	{
+	public function move(
+		Language|string $fromLanguage,
+		VersionId $toVersionId,
+		Language|string $toLanguage
+	): void {
 		$this->ensure($fromLanguage);
-		$this->model->storage()->move($this->id, $fromLanguage, $toVersionId, $toLanguage);
+		$this->model->storage()->move(
+			fromVersionId: $this->id,
+			fromLanguage: $this->language($fromLanguage),
+			toVersionId: $toVersionId,
+			toLanguage: $this->language($toLanguage)
+		);
 	}
 
 	/**
 	 * Returns the stored content fields
 	 *
-	 * @param string $lang Code `'default'` in a single-lang installation
 	 * @return array<string, string>
+	 *
+	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
 	 */
-	public function read(string $language = 'default'): array
+	public function read(Language|string $language = 'default'): array
 	{
 		$this->ensure($language);
```

Comment:
`Idea`: I wonder whether we should replace this `ensure` call with a nullable return value.

Because with the current logic, there could be race conditions. E.g. the calling code calls `exists()` and on success it calls `read()`. But if the version/language is deleted or moved in between those calls, the `read()` call will throw an unexpected exception. If `read` would return `null` in this case, the calling code could handle that much more gracefully without having to wrap everything in a `try catch`.

But then, we have the same potential issue in `move()`, `touch()` and `update()` and I'm not sure how we would solve it there.

Reply:
I haven't added it to the outlook section so far because it's in the steps after PR 10, but I ran into more situations where `::read` actually should return an empty array if the version does not exist or cannot be read. 

I would vote to solve this in a later step when we see why it needs to be different.

Reply:
Empty array is confusing/inconsistent IMO as it could well be that the version exists but does not have fields. But `null` could work.

Fine with doing it later. Could you please add it as a todo?